### PR TITLE
feat: script and predicate blobs

### PR DESCRIPTION
**Pull request by segfault-magnet**
closes: #1519 

Waiting for a fuel-core release to remove all the patches in Cargo.toml

# Release notes

In this release, we:

- Implemented support for pre-uploading code for scripts and predicates via blobs.
# Summary

Added a `Executable` type which can be given a Sway binary and be used to turn it into loader code. 

All script/predicate code except their data sections is uploaded as a blob and referenced in the loader code. 

Every loader code execution can be configured differently without the need of large transactions due to the rest of the code being uploaded beforehand as a blob.

# Checklist

- [x] All **changes** are **covered** by **tests** (or not applicable)
- [x] All **changes** are **documented** (or not applicable)
- [x] I **reviewed** the **entire PR** myself (preferably, on GH UI)
- [x] I **described** all **Breaking Changes** (or there's none)

### CODE REVIEW

File: docs/src/preuploading-code.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+```
+
+Once you want to execute the predicate, you must beforehand upload the blob
+containing its code:
+
+```rust,ignore
+{{#include ../../e2e/tests/predicates.rs:uploading_the_blob}}
+```
+
+By pre-uploading the predicate code you allow for cheaper calls to the predicate
```
Comment (digorithm) [inline docs/src/preuploading-code.md:41]: Grammar nit.

```suggestion
By pre-uploading the predicate code, you allow for cheaper calls to the predicate
```

File: docs/src/preuploading-code.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+## Scripts
+
+A high level pre-upload:
+
+```rust,ignore
+{{#include ../../e2e/tests/scripts.rs:preload_high_level}}
+```
+
+The upload of the blob is handled inside of `convert_into_loader` method. If you
```
Comment (digorithm) [inline docs/src/preuploading-code.md:18]:
```suggestion
The upload of the blob is handled inside of the `convert_into_loader` method. If you
```

File: docs/src/preuploading-code.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+want more fine grained control over it you can create the script transaction
```
Comment (digorithm) [inline docs/src/preuploading-code.md:19]:
```suggestion
want more fine-grained control over it, you can create the script transaction
```

File: e2e/tests/contracts.rs [hunk 4]
```diff
@@ -1476,6 +1478,7 @@ async fn test_contract_submit_and_response() -> Result<()> {
     let contract_methods = contract_instance.methods();
 
     let submitted_tx = contract_methods.get(1, 2).submit().await?;
+    tokio::time::sleep(Duration::from_millis(500)).await;
```
Comment (digorithm) [inline e2e/tests/contracts.rs:1481]: What's with these sleep statements over here?

**Review by hal3e — DISMISSED** [review]
Awesome work! very clean and well written.

File: e2e/tests/contracts.rs (hunk 4 above)
Comment (segfault-magnet) [reply e2e/tests/contracts.rs:1481]: Flakiness when running the tests against the new core binary. If you ask for the transaction status too soon after you submitted it, the core will complain about the transaction not existing.

**Review by Br1ght0ne — DISMISSED** [review]
LGTM, as I saw most of it being written ;)

**Review by MujkicA — APPROVED** [review]

**Review by digorithm — APPROVED** [review]

**Review by hal3e — APPROVED** [review]

**Review by hal3e — APPROVED** [review]

**Review by MujkicA — APPROVED** [review]

**Review by digorithm — APPROVED** [review]

**Review by Br1ght0ne — APPROVED** [review]

**Pull request merged**